.github/workflows/docs.yml

```diff
@@ -11,63 +11,167 @@
+
     runs-on: ubuntu-latest
     permissions:
       pull-requests: write
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4.2.2
+
       - name: Setup Bazel
         uses: bazel-contrib/setup-bazel@0.9.1
+
       - name: Build documentation
         run: |
           bazel build //docs:github-pages && cp bazel-bin/docs/github-pages.tar .
-      - name: Upload artifact for deployment
-        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
-        uses: actions/upload-artifact@v4.4.0
-        with:
-          name: github-pages
-          path: ${{ github.workspace }}/github-pages.tar
-          retention-days: 1
-          if-no-files-found: error
-      - name: Upload artifact for pull request link
+
+      - name: Upload artifact for job analysis
         if: github.event_name == 'pull_request'
-        id: upload-artifact-pr
         uses: actions/upload-artifact@v4.4.0
         with:
           name: github-pages-${{ github.sha }}
-          path: ${{ github.workspace }}/github-pages.tar
+          path: github-pages.tar
           retention-days: 1
           if-no-files-found: error
-      - name: Comment artifact URL
-        if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false
-        run: |
-          gh pr comment ${{ github.event.pull_request.number }} \
-            --body "Documentation artifact: ${{ steps.upload-artifact-pr.outputs.artifact-url }}"
-        env:
-          GH_TOKEN: ${{ github.token }}
+
   docs-deploy:
+    name: Deploy documentation to GitHub Pages
+    description: |
+      Deploy the previously built documentation to GitHub Pages, possibly with a versioned URL.
+      The job will:
+      * update the versions file in the gh-pages branch.
+      * deploy the documentation to the gh-pages branch to an appropriate folder.
+
+    if: ${{ (github.event_name != 'pull_request' || github.event.action != 'closed') }}
+
     environment:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
     permissions:
       pages: write
       id-token: write
     runs-on: ubuntu-latest
-    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
     needs: docs-build
+
+    steps:
+      - name: Determine target folder for github pages
+        # TBD: following typical conventions the main branch should be deployed
+        #      to `/latest` instead of `/`?!
+        run: |
+          if [[ ${{github.event_name}} == 'pull_request' ]]; then
+            echo "target_folder=pr-${{github.event.pull_request.number}}" >> $GITHUB_ENV
+          elif [[ ${{github.ref_name}} != 'main' ]]; then
+            echo "target_folder=${{github.ref_name}}" >> $GITHUB_ENV
+          else
+            echo "target_folder=/" >> $GITHUB_ENV
+          fi
+
+      # TBD: Maybe we should not split the build and deploy jobs, but rather have a single job?
```

Comment:
When merge queue is introduced you will want to skip the whole deploy job.

Reply:
Ok, I'll keep it separated.